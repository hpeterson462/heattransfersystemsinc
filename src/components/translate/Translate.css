.translate-container {
  display: flex;
  justify-content: center;
  margin: 5rem;
  margin-top: 10rem;
  margin-bottom: 10rem;
}
