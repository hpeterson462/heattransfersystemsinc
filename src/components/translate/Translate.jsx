import React, { useEffect } from 'react';
import './Translate.css';

export default function Translate() {

  const googleTranslateElementInit = () => {
    new window.google.translate.TranslateElement({ pageLanguage: 'en' }, 'google_translate_element')
  }

  useEffect(() => {
    const addScript = document.createElement('script');
    addScript.setAttribute('src', '//translate.google.com/translate_a/element.js?cb=googleTranslateElementInit');
    document.body.appendChild(addScript);
    window.googleTranslateElementInit = googleTranslateElementInit;
  }, [])

  return (
    <div className="translate-container">
      <div id="google_translate_element"></div>
    </div>
  )
}
