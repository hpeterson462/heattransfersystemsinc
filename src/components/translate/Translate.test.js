import React from 'react';
import { cleanup, render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Translate from './Translate';

describe('Translate component', () => {
  afterEach(() => cleanup());

  it('renders Translate component', () => {
    const { asFragment } = render(<MemoryRouter><Translate /></MemoryRouter>);

    expect(asFragment()).toMatchSnapshot();
  });
});
