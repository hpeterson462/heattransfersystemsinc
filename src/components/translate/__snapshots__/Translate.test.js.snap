// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Translate component renders Translate component 1`] = `
<DocumentFragment>
  <div
    class="translate-container"
  >
    <div
      id="google_translate_element"
    />
  </div>
</DocumentFragment>
`;
